packages/kbn-scout/src/playwright/page_objects/maps_page.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { retryForSuccess } from '@kbn/ftr-common-functional-services';
+import { ToolingLog } from '@kbn/tooling-log';
+import { ScoutPage } from '../fixtures/types';
+
+const RENDER_COMPLETE_SELECTOR = '[data-render-complete="true"]';
+const RENDER_COMPLETE_PENDING_SELECTOR = '[data-render-complete="false"]';
+const DATA_LOADING_SELECTOR = '[data-loading]';
+
+const debugLog = ToolingLog.bind(ToolingLog, { level: 'debug', writeTo: process.stdout });
+export class MapsPage {
+  constructor(private readonly page: ScoutPage) {}
+
+  async goto() {
+    await this.page.gotoApp('maps');
+  }
+
+  async waitForRenderCompletion(selector: string = RENDER_COMPLETE_SELECTOR) {
+    // This is my first attempt at a simple solution for test/functional/services/renderable.ts#waitForRender()
+    await this.page.locator(selector).waitFor();
+  }
+  async waitForRender(count: number = 1): Promise<void> {
+    await retryForSuccess(new debugLog({ context: 'MapsPage' }), {
+      retryCount: 10,
+      retryDelay: 1500,
+      timeout: 10_000,
+      methodName: 'waitForRender()',
+      block: async () => {
+        const renderCompleteLocator = this.page.locator(RENDER_COMPLETE_SELECTOR);
+        const renderPendingDataTitleLocator = this.page
+          .locator(RENDER_COMPLETE_PENDING_SELECTOR)
+          .and(this.page.locator('data-title'));
```

Comment:
why do we need to combine locator here? In original code we fetch `data-title` attribute for pending elements, but search only for `'[data-render-complete="false"]'`

Reply:
So, pending elements comes from
```
        const pendingElements = await this.find.allByCssSelector(
          RENDER_COMPLETE_PENDING_SELECTOR,
          2500
        );
```
Then from all the pending elements, we build a list of titles right?
Is the combined locator not doing the same thing?

Reply:
Probably yes, I thought you gonna tell me how it really should work :)

Reply:
It seems to be working as is. :)